### PR TITLE
Switch to crossterm backend, async input

### LINKED ISSUE #71: keyboard input sluggish for slow ssh connections
On very slow connections the app sometimes take several seconds to process keyboard input.
This is is especially a problem when many cells are updated during drawing, IE when showing an rgb stream of a moving robot.

Can be reproduced conveniently with trickle:
`trickle -d 400 -u 400 -s -t 0.1 ssh user@host`

### PR DESCRIPTION
**Pull request by carzum**
Fixes #71

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -38,18 +43,18 @@ fn main() -> Result<(), Box<dyn Error>> {
...
 
-    let mut key_to_input: HashMap<Key, String> = conf
+    let mut key_to_input: HashMap<KeyCode, String> = conf
         .key_mapping
         .iter()
         .map(|(v, k)| match k.as_str() {
-            "Enter" => (Key::Char('\n'), v.clone()),
-            "Esc" => (Key::Esc, v.clone()),
-            _ => (Key::Char(k.chars().next().unwrap()), v.clone()),
+            "Enter" => (KeyCode::Backspace, v.clone()),
```
Comment (Eman7C7) [inline src/main.rs:50]: Backspace as Enter?

**Review by Eman7C7 — CHANGES_REQUESTED** [review]
All good except the backspace thing

**Review by Eman7C7 — APPROVED** [review]

File: src/main.rs [hunk 2]
```diff
@@ -74,34 +79,55 @@ fn main() -> Result<(), Box<dyn Error>> {
...
-                        running_app.handle_input(&key_to_input[&input]);
-                    } else {
-                        running_app.handle_input(&app_modes::input::UNMAPPED.to_string());
-                    }
-                }
-            },
-            Event::Tick => {}
-        }
     }
+    // restore terminal
```
Comment (MaxiMaerz) [inline src/main.rs:124]: nice

**Review by MaxiMaerz — APPROVED** [review]

File: src/main.rs [hunk 3]
```diff
@@ -11,18 +10,24 @@ mod marker;
...
+    terminal::{disable_raw_mode, LeaveAlternateScreen},
+};
 use rosrust;
-use rustros_tf;
+use rustros_tf::TfListener;
 use std::error::Error;
-use termion::event::Key;
 
-fn main() -> Result<(), Box<dyn Error>> {
+#[tokio::main]
```
[the remark this answers is not in the document]
Comment (carzum) [reply src/main.rs:29]: the macro allows the function to be async. https://tokio.rs/tokio/tutorial/hello-tokio

**Pull request merged**